src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -304,13 +304,24 @@ def init_sync_d_lag_tables(db_conn):
     if_name_lag_name_map = {}
     # { OID -> lag_name (SONiC) }
     oid_lag_name_map = {}
+    # { lag_name (SONiC) -> lag_oid (SAI) }
+    lag_sai_map = {}
+    # { lag_oid (SAI) -> lag_name (SONiC) }
+    sai_lag_map = {}
 
     db_conn.connect(APPL_DB)
 
     lag_entries = db_conn.keys(APPL_DB, "LAG_TABLE:*")
 
     if not lag_entries:
-        return lag_name_if_name_map, if_name_lag_name_map, oid_lag_name_map
+        return lag_name_if_name_map, if_name_lag_name_map, oid_lag_name_map, lag_sai_map
+
+    db_conn.connect(COUNTERS_DB)
+    lag_sai_map = db_conn.get_all(COUNTERS_DB, "COUNTERS_LAG_NAME_MAP")
+    for name, sai_id in lag_sai_map.items():
+        sai_id_key = get_sai_id_key(db_conn.namespace, sai_id.lstrip("oid:0x"))
```

Comment:
Can we use if_lag_name_map from port_util.if_lag_name_map() ?

Reply:
The function `get_interface_oid_map` returns both port and lag. So not fit well.

---
In reply to: [594597081](https://github.com/Azure/sonic-snmpagent/pull/203#discussion_r594597081) [](ancestors = 594597081)